langchain4j-open-ai-official/src/main/java/dev/langchain4j/model/openaiofficial/InternalOpenAiOfficialHelper.java

```
@@ -0,0 +1,602 @@
+        };
+    }
+
+    static Response<AiMessage> convertResponse(ChatResponse chatResponse) {
+        return Response.from(
+                chatResponse.aiMessage(),
+                chatResponse.metadata().tokenUsage(),
+                chatResponse.metadata().finishReason());
+    }
+
+    static StreamingChatResponseHandler convertHandler(StreamingResponseHandler<AiMessage> handler) {
+        return new StreamingChatResponseHandler() {
+
+            @Override
+            public void onPartialResponse(String partialResponse) {
+                handler.onNext(partialResponse);
+            }
+
+            @Override
+            public void onCompleteResponse(ChatResponse completeResponse) {
+                handler.onComplete(convertResponse(completeResponse));
+            }
+
+            @Override
+            public void onError(Throwable error) {
+                handler.onError(error);
+            }
+        };
+    }
+
+    static void validate(ChatRequestParameters parameters) {
+        if (parameters.topK() != null) {
+            throw new UnsupportedFeatureException("'topK' parameter is not supported by OpenAI");
+        }
+    }
+
+    static ResponseFormat fromOpenAiResponseFormat(String responseFormat) {
+        if ("json_object".equals(responseFormat)) {
+            return JSON;
+        } else {
+            return null;
+        }
+    }
+
+    static ChatCompletionCreateParams.Builder toOpenAiChatCompletionCreateParams(
+            ChatRequest chatRequest,
+            OpenAiOfficialChatRequestParameters parameters,
+            Boolean strictTools,
+            Boolean strictJsonSchema) {
+
+        // OpenAI-specific parameters
+        ChatCompletionCreateParams.Builder builder =
+                ChatCompletionCreateParams.builder().model(parameters.modelName());
+
+        if (parameters.maxOutputTokens() != null && parameters.maxCompletionTokens() == null) {
```

```suggestion
        if (parameters.maxOutputTokens() != null) {
```
Why do we need this additional check?

---

That's because you can't set them both (the older API and the new API together isn't a good idea), so you would get this error:
```
com.openai.errors.BadRequestException: 400: OpenAIError{additionalProperties={error={message=Setting 'max_tokens' and 'max_completion_tokens' at the same time is not supported., type=invalid_request_error, param=max_tokens, code=invalid_parameter_combination}}}
```
And then I set precedence for the new API.

---

So if user sets both `maxOutputTokens` and `maxCompletionTokens`, it will silently ignore `maxOutputTokens`? I would let the SDK/API validate it and throw an error instead of doing this (unobvious) trick here